cgyle/catalog.py

```diff
@@ -121,19 +121,25 @@ def translate_policy(
     ) -> List[str]:
         result: List[str] = []
         glob_regexp_map = {
-            '*': '[^/]+',
-            '**': '.+'
+            '/*': '/[^/]*',
+            '**': '.*',
+            '*': '[^/]*'
         }
         with open(policy_file) as policy:
             policy_dict = yaml.safe_load(policy)
             for category in policy_dict:
                 if category not in skip_sections:
                     for pattern in policy_dict.get(category):
-                        pattern = pattern.replace(
-                            '**', glob_regexp_map.get('**')
-                        )
-                        pattern = pattern.replace(
-                            '*', glob_regexp_map.get('*')
-                        )
+                        if pattern == '*':
+                            pattern = pattern.replace(
+                                '*', glob_regexp_map.get('*')
+                            )
+                        else:
+                            pattern = pattern.replace(
+                                '**', glob_regexp_map.get('**')
+                            )
+                            pattern = pattern.replace(
+                                '/*', glob_regexp_map.get('/*')
+                            )
```

Comment:
I don't think that's ok.
The asterisk in the policy `bar*` will not be replaced because it neither is a single `*` nor matches `/*`.
So it remains `bar*` which is a valid pattern for `ba` and zero-or-more `r`s.
That also matches the repo `bar` and so the test succeeds although the pattern is not right.

Reply:
Just add `bat` to the repo list. It should not be matched by `bar*`, but it will be matched by the current code.

Reply:
yes good catch

Reply:
Ok, how about it now ?